src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/SeedJob.kt

```diff
@@ -27,4 +28,18 @@ abstract class SeedJob<RowType>(
   abstract fun processRow(row: RowType)
 }
 
+abstract class ExcelSeedJob(
+  val fileName: String,
+  val premisesId: UUID,
+  val sheetName: String,
+) {
+  init {
+    if (fileName.contains("/") || fileName.contains("\\")) {
+      throw RuntimeException("Filename must be just the filename of a .xlsx file in the /seed directory, e.g. for /seed/upload.xlsx, just `upload` should be supplied")
+    }
+  }
+
+  abstract fun processDataFrame(dataFrame: DataFrame<*>)
```

Comment:
Again, one for once everything is working but i think seed jobs should be spring managed beans and whatever arguments are required should be passed into the function doing the processing. I'm slowly working on changing the existing seed jobs to use this pattern

Reply:
This has been changed for all seed jobs now.  Have followed the same principle for this job.